lr_face/evaluators.py

```diff
@@ -154,22 +154,47 @@ def calculate_metrics_dict(scores, y, lr_predicted, label):
             'accuracy' + label: accuracy_score(y, scores > .5)}
 
 
-def evaluate(lr_system: CalibratedScorer,
-             test_pairs: List[FacePair],
+def evaluate(lr_systems: Dict[Tuple, CalibratedScorer],
+             test_pairs_per_category: Dict[Tuple, List[FacePair]],
              make_plots_and_save_as: Optional[str]) -> Dict[str, float]:
     """
     Calculates a variety of evaluation metrics and plots data if
     `make_plots_and_save_as` is not None.
     """
-    scores = lr_system.scorer.predict_proba(test_pairs)[:, 1]
-    lr_predicted = lr_system.calibrator.transform(scores)
-    y_test = [int(pair.same_identity) for pair in test_pairs]
 
-    if make_plots_and_save_as:
-        calibrator = lr_system.calibrator
-        if type(calibrator) == ELUBbounder:
-            calibrator = calibrator.first_step_calibrator
+    scores = np.array([])
+    lr_predicted = np.array([])
+    y_test = []
+    test_pairs = []
+    for category, pairs in test_pairs_per_category.items():
+        if category not in lr_systems:
+            print(f'skipping {pairs} for category {category}')
+            continue
+        category_scores = lr_systems[category].scorer.predict_proba(pairs)[
+                          :, 1]
+        scores = np.append(scores, category_scores)
+        lr_predicted = np.append(
+            lr_predicted,
+            lr_systems[category].calibrator.transform(category_scores))
+        category_y_test = [int(pair.same_identity) for pair in pairs]
+        y_test += category_y_test
+        test_pairs += pairs
+        if make_plots_and_save_as:
+            calibrator = lr_systems[category].calibrator
+            if type(calibrator) == ELUBbounder:
+                calibrator = calibrator.first_step_calibrator
+            plot_score_distribution_and_calibrator_fit(
+                calibrator,
+                category_scores,
+                category_y_test,
+                savefig=f'{make_plots_and_save_as} {category} '
```

Comment:
Bij mij klapt hij er hier soms wel/ soms niet uit op een assertion error in lir. expected two classes; found: [0].

Reply:
dat had ik ondervangen, dacht ik. Krijg je dit op de meest recente versie?

Reply:
bij mij doet hij dit ook af en toe, op inderdaad de meest recente versie (vanochtend, ik weet niet wat je daarna nog hebt veranderd)